core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -418,6 +425,64 @@ public final SegmentsStats segmentsStats() {
                 stats.addTermVectorsMemoryInBytes(guardedRamBytesUsed(segmentReader.getTermVectorsReader()));
                 stats.addNormsMemoryInBytes(guardedRamBytesUsed(segmentReader.getNormsReader()));
                 stats.addDocValuesMemoryInBytes(guardedRamBytesUsed(segmentReader.getDocValuesReader()));
+
+                if (!detailed) {
```

Comment:
Can you factor out all the new code into a new private method `getFileSizeStats` or something, and then invert this if statement to `if (detailed) {...}`?